src/Symfony/Component/Workflow/DependencyInjection/WorkflowEventListenerExtractorPass.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\ErrorHandler\ErrorRenderer\FileLinkFormatter;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Workflow\Definition;
+use Symfony\Component\Workflow\EventListener\GuardExpression;
+use Symfony\Component\Workflow\EventListener\GuardListener;
+use Symfony\Component\Workflow\Transition;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class WorkflowEventListenerExtractorPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition('data_collector.workflow')) {
+            return;
+        }
+
+        $dispatcher = $container->get('event_dispatcher', ContainerBuilder::NULL_ON_INVALID_REFERENCE);
```

Comment:
Doing a `->get()` call in a compiler pass is a no-go. In compiler passes, services are not yet ready to be instantiated as the configuration is not complete yet.

Reply:
doing it at compile time requires doing it based only on service definitions.

However, as the event dispatcher has all its listeners registered at compile time, you could inspect the definition of listeners registered in it instead of using the runtime API (inspect the calls to `addListener` to identify the added listeners, and handle the case of adding a listener using `[new ServiceClosureArgument(new Reference('service id')), 'method']` to identify the ReflectionMethod for such listener).